phac-nml/irida-next
# Standardize policy methods

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Updated authorization methods in policies to be standardized so we can provide better messages to the user when they can't access a resource or perform an action.

Standardized policy authorization methods to the following between the policy classes

Note: We can modify the policy authorization messages as we go but this should be good for now.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**When trying to access a group which a user doesn't have access as well as it's associations (samples, members, settings):**

![image](https://github.com/phac-nml/irida-next/assets/25466211/3e4da2a1-4a81-4bd5-a5da-37bc1bf41f19)

![image](https://github.com/phac-nml/irida-next/assets/25466211/a26642e5-0ae0-4684-8180-b0963d46bdad)

**When trying to access a project which a user doesn't have access as well as it's associations (samples, members, settings):**

![image](https://github.com/phac-nml/irida-next/assets/25466211/57d07e95-a58a-45e7-8056-9d9ae0a0142f)

![image](https://github.com/phac-nml/irida-next/assets/25466211/713a39bf-7b70-469b-b6d2-8304380a5165)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1) Login as `admin@email.com`
2) Go to the groups tab
3) Click on the group `Bacillus`
4) Make a note of the urls for the group, the members, samples, and settings
5) Logout and login as `user3@email.com`
6) Try to visit the urls from step 4. You should be presented with the group access error screens as in the screenshots section above
7) Repeat steps 1 through 6 but this time do the same for projects. You should be presented with the project access error screens as in the screenshots section above
